ror-tools/src/main/scala/tech/beshu/ror/tools/RorToolsApp.scala

```diff
@@ -16,43 +16,82 @@
+    }
+  }
+
+  def handle(args: Array[String]): Result = {
+    OParser.parse(parser, args, Config(Command.Verify(None), UserUnderstandsImplicationsOfESPatching.AnswerNotGiven)) match {
+      case None =>
+        Result.CommandNotParsed
+      case Some(config) =>
         Try {
           config.command match {
             case Command.Patch(customEsPath) =>
-              val esDirectory = esDirectoryFrom(customEsPath)
-              new PatchAction(EsPatch.create(esDirectory)).execute()
+              config.userUnderstandsImplicationsOfESPatching match {
+                case UserUnderstandsImplicationsOfESPatching.Yes =>
+                  performPatching(customEsPath)
+                case UserUnderstandsImplicationsOfESPatching.No =>
+                  patchingAbortedBecauseUserDidNotAcceptConsequences()
+                case UserUnderstandsImplicationsOfESPatching.AnswerNotGiven =>
+                  if (askForConfirmation() == "yes") performPatching(customEsPath)
+                  else patchingAbortedBecauseUserDidNotAcceptConsequences()
+              }
             case Command.Unpatch(customEsPath) =>
               val esDirectory = esDirectoryFrom(customEsPath)
               new UnpatchAction(EsPatch.create(esDirectory)).execute()
+              Result.Success
             case Command.Verify(customEsPath) =>
               val esDirectory = esDirectoryFrom(customEsPath)
               new VerifyAction(EsPatch.create(esDirectory)).execute()
+              Result.Success
           }
-        } recover {
-          case ex: RorToolsException =>
-            println(s"ERROR: ${ex.printStackTrace()}")
-            sys.exit(1)
-          case ex: Throwable =>
-            println(s"UNEXPECTED ERROR: ${ex.printStackTrace()}")
-            sys.exit(1)
+        } match {
+          case Failure(ex: RorToolsException) =>
+            println(s"ERROR: ${ex.getMessage()}\n${ex.printStackTrace()}")
+            Result.Failure(1)
+          case Failure(ex: Throwable) =>
+            println(s"UNEXPECTED ERROR: ${ex.getMessage()}\n${ex.printStackTrace()}")
+            Result.Failure(1)
+          case Success(result) =>
+            result
         }
-      }
+    }
+  }
+
+  private def patchingAbortedBecauseUserDidNotAcceptConsequences(): Result = {
+    println("You have to confirm, that You understand the implications of ES patching in order to perform it.\nYou can read about patching in our documentation: https://docs.readonlyrest.com/elasticsearch#id-3.-patch-elasticsearch.")
+    Result.Failure(1)
+  }
+
+  private def askForConfirmation(): String = {
+    println("Elasticsearch needs to be patched to work with ReadonlyREST. You can read about patching in our documentation: https://docs.readonlyrest.com/elasticsearch#id-3.-patch-elasticsearch.")
+    print("Do you understand the implications of ES patching? (yes/no): ")
+    StdIn.readLine()
+  }
+
+  private def performPatching(customESPath: Option[Path]): Result = {
```

Comment:
shouldn't we add similar methods from unpatch and verify? to be consistent and for the sake of redability